File: reversing_array.py
def reverse_array(arr):
    """
    Reverses the order of elements in an array.

    Parameters:
    arr (list): The array to be reversed.

    Returns:
    list: The reversed array.
    """
    # Use slicing to reverse the array
    reversed_arr = arr[::-1]
    return reversed_arr

# Example usage:
if __name__ == "__main__":
    original_array = [1, 2, 3, 4, 5]
    reversed_array = reverse_array(original_array)
    print("Original array:", original_array)
    print("Reversed array:", reversed_array)

File: reversing_array.java
public class ArrayReversal {
    public static void main(String[] args) {
        int[] arr = {1, 2, 3, 4, 5};
        reverseArray(arr);
        System.out.println("Reversed array: " + java.util.Arrays.toString(arr));
    }

    public static void reverseArray(int[] arr) {
        /*
         * Reverses the given array in-place by swapping elements.
         *
         * Parameters:
         * arr (int[]): The array to be reversed.
         */
        int start = 0;
        int end = arr.length - 1;

        while (start < end) {
            // Swap the elements at the start and end positions
            int temp = arr[start];
            arr[start] = arr[end];
            arr[end] = temp;

            start++;
            end--;
        }
    }
}
